### ISSUE TITLE
Radiant Amphora + Ender IO Fluid Tank -> Infinite Fluid duplication

### ISSUE THREAD

**Issue by al132mc** [AranaiRa/ArcaneArchives#68]
I'm unsure if this is a bug on your end or on theirs (or both?), but I figured that i'd post it here first. 

[Demonstration](https://streamable.com/u9tce)

A specific description of the issue by one of my users can be found here https://github.com/al132mc/Glacial-Awakening/issues/52

**Comment by duely** [comment]
I'm not entirely sure whose side that is on... It's going to be a matter of replicating it and tracing the function calls to see why it's extracting fluid in simulate but not actually extracting them (if that's what it is)...

**Comment by duely** [comment]
After looking at the video, just to double check, is this happening in a single-player or a server world?

**Comment by duely** [comment]
Disregard, I've replicated it on a server.

**Comment by duely** [comment]
So, the word from them is basically "filling up your item stack shouldn't have any real-world consequences it's your fault". I can see where they're coming from, and especially given the fact that they can't actually *drain* it once filled, I could get their point -- if they actually attempted to drain the stack after simulating.

The basic issue is that they perform a simulation of copying the itemstack, filling it up but setting "simulate" to false, and then if the resulting item doesn't stack with the item in the slot, they discard it and keep trying.

If I modified the amphora's mode to be ignored, and it were drained before being discarded, that would help. Likewise, I'm not sure why they even begin the simulation of filling the itemstack in the first place when the items clearly don't match.

Regardless, I've just asked them to provide a blacklist functionality if that's the way they want to do things. Once I formulate a better/saner response, I'll dive back into the thread.

**Comment by duely** [comment]
While part of the blame definitely lies on our shoulders, there's no clear indication that an ItemStack *must* be copyable and editable without it having an effect on the rest of the world. This situation occurs because of the "queue" input and output slots of the Tank and wouldn't happen with any other device unless they did something similar.

My only recommendation would be to disable the EnderiO Tank if you are concerned that fluid duplication might be a thing. 

As the only other solution would be to disable the Amphora's ItemStack capabilities entirely, which would affect way more than just the EnderIO tank, I'm really at a loss. There's been no further replies in the past three weeks since I raised the issue on their tracker, and although I've discussed with tterrag an idea surrounding Mix-Ins, it occurs to me that I don't want to insert ASM into our code specifically to deal with the fact that the interface (however long it's been around) for the EnderIO tank is ridiculous. 

Hence, I'm closing this issue. I'll link to it from known issues at a point in the future.
